modules/core/arrow-extras-data/src/main/kotlin/arrow/data/fingertree/internal/Affix.kt

```diff
@@ -0,0 +1,35 @@
+package arrow.data.fingertree.internal
+
+internal sealed class Affix<A> {
+  data class One<A>(val a: A) : Affix<A>()
+  data class Two<A>(val a: A, val b: A) : Affix<A>()
+  data class Three<A>(val a: A, val b: A, val c: A) : Affix<A>()
+  data class Four<A>(val a: A, val b: A, val c: A, val d: A) : Affix<A>()
+
+  fun append(item: A): Affix<A> = fromList(this.toList() + listOf(item))
```

Comment:
I'm sure this can be made faster by removing the additional intermediate list of items `this.toList() + listOf(item)`